application/Modules/ClipboardTest/ClipboardModuleTest.cs

```diff
@@ -0,0 +1,64 @@
+using System.ComponentModel.Composition;
+using System.ComponentModel.Composition.Hosting;
+using System.Diagnostics;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using MORR.Modules.Clipboard;
+using MORR.Modules.Clipboard.Producers;
+
+namespace ClipboardTest
+{
+    [TestClass]
+    public class ClipboardModuleTest
+    {
+        private Mock<ClipboardCopyEventProducer> clipboardCopyEventProducer;
+        private Mock<ClipboardCutEventProducer> clipboardCutEventProducer;
+        private ClipboardModule clipboardModule;
+        private Mock<ClipboardPasteEventProducer> clipboardPasteEventProducer;
+        private CompositionContainer container;
+
+        [TestInitialize]
+        public void BeforeTest()
+        {
+            clipboardModule = new ClipboardModule();
+            clipboardCopyEventProducer = new Mock<ClipboardCopyEventProducer>();
+            clipboardCutEventProducer = new Mock<ClipboardCutEventProducer>();
+            clipboardPasteEventProducer = new Mock<ClipboardPasteEventProducer>();
+            container = new CompositionContainer();
+            container.ComposeExportedValue(clipboardCopyEventProducer.Object);
+            container.ComposeExportedValue(clipboardCutEventProducer.Object);
+            container.ComposeExportedValue(clipboardPasteEventProducer.Object);
+            container.ComposeParts(clipboardModule);
+        }
+
+        [TestMethod]
+        public void TestClipboardModule_Activate()
+        {
+            // Preconditions
+            Debug.Assert(clipboardModule != null);
+
+            /* GIVEN */
+
+            /* WHEN */
+            clipboardModule.IsActive = true;
```

Comment:
I don't know if it's the same for you, but this throws an Exception for me as `GlobalHook` fails to load the .dll (dunno why that is, probably botched working directory? It works fine for me when i launch the CLI.exe from any working directory however.)

If the module would really use the Mocks for the producers, should this even happen (i'm not familiar with the Moq API)?

Reply:
The problem with this is that the output directory of the test projects is not the same as that of the other assemblies. You will have to manually copy the hook dlls to the corresponding bin folder inside the directory corresponding to the test projects.
Regarding the second point: I think Moq doesn't do anything by default (and here it seems the objects are only wrapped in the Mock object, but there are not indirections added), which would be why you would still see the problem